packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -1362,8 +1371,51 @@ void main() {
     await tester.testTextInput.receiveAction(TextInputAction.next);
     await tester.pump();
 
-    // Still has focus after pressing "next".
-    expect(focusNode.hasFocus, true);
+    // Next node has focus after pressing 'next' button.
+    expect(focusNode.hasFocus, isFalse);
+    expect(Focus.of(previousKey.currentContext).hasFocus, isFalse);
+    expect(Focus.of(nextKey.currentContext).hasFocus, isTrue);
+  });
+
+  testWidgets('Moves focus to the previous field by default when "previous" action is pressed', (WidgetTester tester) async {
```

Comment:
In theory, you could do all of this checking in just one slightly more complicated version of this test.

Reply:
Well, sure, but doesn't that violate the principle of "one behavior => one test"?

Reply:
OK, I modified the tests to test all of the behaviors in separate tests (so they fail separately), but using a new `TestVariant` subclass so that I don't duplicate code when doing so. Let me know what you think.